packages/react-native/processSourceMap.js

```diff
@@ -0,0 +1,62 @@
+const bundleToString = require('metro/src/lib/bundleToString');
+const baseJSBundle = require('metro/src/DeltaBundler/Serializers/baseJSBundle');
+const CountingSet = require('metro/src/lib/CountingSet').default;
+const fs = require('fs');
+const path = require('path');
+const { DebugIdGenerator, SourceProcessor } = require('@backtrace/sourcemap-tools');
+
+const DEBUG_ID_PATH = process.env.DEBUG_ID_PATH;
+
+/**
+ * Process metro build with source map support powered by Backtrace.
+ */
+function processSourceMap(entryPoint, preModules, graph, options) {
+    const bundle = bundleToString(baseJSBundle(entryPoint, preModules, graph, options));
+
+    // development build - skip source map upload
+    if (graph.transformOptions.hot || graph.transformOptions.dev) {
+        return bundle;
+    }
+
+    // no source map option set
+    const sourceMapOutputPathParameter = process.argv.indexOf('--sourcemap-output');
+    if (sourceMapOutputPathParameter === -1) {
+        return bundle;
+    }
+
+    const sourceMapOutputPath = process.argv[sourceMapOutputPathParameter + 1];
+    const backtraceSourceMapId =
+        DEBUG_ID_PATH ?? path.join(path.dirname(sourceMapOutputPath), '.backtrace-sourcemap-id');
+
+    const debugIdGenerator = new DebugIdGenerator();
+    const sourceProcessor = new SourceProcessor(debugIdGenerator);
+    const { source, debugId } = sourceProcessor.processSource(bundle.code);
+    const snippet = debugIdGenerator.generateSourceSnippet(debugId);
+
+    console.debug(`Backtrace: saving debugId ${debugId} to ${backtraceSourceMapId}.`);
+    fs.writeFileSync(backtraceSourceMapId, debugId);
```

Comment:
small nit: this can be an async function, you could use promise-based functions here

Reply:
async won't give us anything in this case. In my opinion this can cause additional headache and unexpected issues. This is a simple, fast and small operation. I would prefer to stay in the sync context and don't risk on executing a different part of the code faster than we wrote a file.

Reply:
`customSerializer` is awaited in Metro, so there's no chance of that happening. I don't see any risk here.

Reply:
but at the same time - whats the point, what benefits it brings - the build system won't move, we need to block it anyway

Reply:
I guess none, just a matter of styling preference.